docs/src/miden_library.md

```diff
@@ -0,0 +1,128 @@
+
+## Output Note Procedures (`miden::output_note`)
+
+Output note procedures can be used to fetch data on output notes created by the transaction.
+
+| Procedure | Description |
+| --- | --- |
+| `get_assets_info` | Returns the information about assets in the output note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[ASSETS_COMMITMENT, num_assets]`<br>Context: Any |
+| `get_assets` | Writes the assets of the output note with the specified index into memory starting at the specified address.<br><br>Inputs: `[dest_ptr, note_index]`<br>Outputs: `[num_assets, dest_ptr, note_index]`<br>Context: Any |
+| `get_recipient` | Returns the recipient of the output note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[RECIPIENT]`<br>Context: Any |
+| `get_metadata` | Returns the metadata of the output note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[METADATA]`<br>Context: Any |
+
+## Transaction Procedures (`miden::tx`)
+
+Transaction procedures manage transaction-level operations including note creation, context switching, and reading transaction metadata.
+
+| Procedure | Description |
+| --- | --- |
+| `get_block_number` | Returns the block number of the transaction reference block.<br><br>Inputs: `[]`<br>Outputs: `[num]`<br>Context: Any |
+| `get_block_commitment` | Returns the block commitment of the reference block.<br><br>Inputs: `[]`<br>Outputs: `[BLOCK_COMMITMENT]`<br>Context: Any |
+| `get_block_timestamp` | Returns the timestamp of the reference block for this transaction.<br><br>Inputs: `[]`<br>Outputs: `[timestamp]`<br>Context: Any |
+| `get_input_notes_commitment` | Returns the input notes commitment hash.<br><br>Inputs: `[]`<br>Outputs: `[INPUT_NOTES_COMMITMENT]`<br>Context: Any |
+| `get_output_notes_commitment` | Returns the output notes commitment hash.<br><br>Inputs: `[]`<br>Outputs: `[OUTPUT_NOTES_COMMITMENT]`<br>Context: Any |
+| `get_num_input_notes` | Returns the total number of input notes consumed by this transaction.<br><br>Inputs: `[]`<br>Outputs: `[num_input_notes]`<br>Context: Any |
+| `get_num_output_notes` | Returns the current number of output notes created in this transaction.<br><br>Inputs: `[]`<br>Outputs: `[num_output_notes]`<br>Context: Any |
+| `create_note` | Creates a new note and returns the index of the note.<br><br>Inputs: `[tag, aux, note_type, execution_hint, RECIPIENT]`<br>Outputs: `[note_idx]`<br>Context: Native & Account |
+| `add_asset_to_note` | Adds the ASSET to the note specified by the index.<br><br>Inputs: `[ASSET, note_idx]`<br>Outputs: `[ASSET, note_idx]`<br>Context: Native |
+| `build_recipient_hash` | Returns the RECIPIENT for a specified SERIAL_NUM, SCRIPT_ROOT, and inputs commitment.<br><br>Inputs: `[SERIAL_NUM, SCRIPT_ROOT, INPUT_COMMITMENT]`<br>Outputs: `[RECIPIENT]`<br>Context: Any |
```

Comment:
Similar to one of the previous comments - maybe this should move to something like `miden::note::utils` module.